ArduCopter/mode.cpp

```diff
@@ -194,6 +198,10 @@ bool Copter::set_mode(Mode::Number mode, ModeReason reason)
     // return immediately if we are already in the desired mode
     if (mode == flightmode->mode_number()) {
         control_mode_reason = reason;
+        // make happy noise
```

Comment:
Not sure about this one.  If you're getting mode changes via mavlink this might come out multiple times if the GCS is sending the command multiple times to ensure the transition in the face of packet loss, or simply because you have multiple telemetry links to the vehicle.

Reply:
qgc will notify you by voice when the flight mode is switched.  I think this change will be handled by gcs.

Reply:
We could qualify this by reason RC to maintain the existing behavior.

Reply:
This now will only beep is the new mode is different to the old mode and the mode reason is different to the last mode reason.